PWGHF/D2H/Tasks/taskCharmPolarisation.cxx

```diff
@@ -661,12 +705,35 @@ struct TaskPolarisationCharmHadrons {
     return true;
   }
 
+  template <typename TRK, typename PART>
+  void searchFirstLevelMother(TRK const& prongTrack, std::vector<int>& idMothers, PART const&)
+  {
+    /// particle associated to the prong track
+    if (!prongTrack.has_mcParticle()) {
+      return;
+    }
+    auto prongParticle = prongTrack.mcParticle();
+    /// leave the vector of mother indices empty if the currect paticle has no mothers
+    if (!prongParticle.has_mothers()) {
+      return;
+    }
+    // loop over the mother particles of the analysed particle
+    for (auto iMother = prongParticle.mothersIds().front(); iMother <= prongParticle.mothersIds().back(); ++iMother) {
+      // if a mother is already present in the vector, do not check it again
+      if (std::find(idMothers.begin(), idMothers.end(), iMother) != idMothers.end()) {
+        continue;
+      }
+      idMothers.push_back(iMother);
+    }
```

Comment:
why do you loop over all of them, if you in the end only use the first one?

Reply:
this is a refuse, at the beginning I was considering to save them all then I changed my mind. Fixed it now (I break immediately after the 1st mother)